src/video/video_sdl.c

```diff
@@ -478,13 +605,28 @@ void Video_Tick(void)
 	SDL_Event event;
 	static bool s_showFPS = false;
 
+#if defined(_WIN32) && defined(WITH_SDL)
+	if (!s_video_initialized) Video_Init();
+#else
 	if (!s_video_initialized) return;
+#endif
 
 	if (g_fileOperation != 0) return;
 
 	if (s_video_lock) return;
 	s_video_lock = true;
 
+#if defined(_WIN32) && defined(WITH_SDL)
+	if (!g_running && s_fullscreen) {
+		if (g_scale_filter == FILTER_HQX) {
+			s_surfaceFlags = SDL_SWSURFACE;
+		} else {
+			s_surfaceFlags = SDL_SWSURFACE | SDL_HWPALETTE;
+		}
+		g_screen_magnification = s_prev_magnification;
+		Video_Resize();
+	}
+#endif
```

Comment:
Workaround: this because of SDL1.2 under Windows, when you Quit the game in fullscreen mode, it freezes. So we first return on windowed mode, then can dispose and close all.